docs/markdown/Adding-new-projects-to-wrapdb.md

```diff
@@ -51,6 +51,73 @@ If the project name is too generic or ambiguous (e.g. `benchmark`),
 consider using `organization-project` naming format (e.g.
 `google-benchmark`).
 
+## Overriding dependencies in the submitted project
+
+Ideally the project you submit should make a call to `meson.override_dependency`
+for each dependency you would like to expose, with the first argument
+matching the pkg-config file name. This abstracts away the
+need to know and keep track of the variable names downstream.
+
+For instance, the Apache Arrow project exposes multiple dependencies like
+its base `arrow` library, along with an `arrow-compute` library. The
+project generates `arrow.pc` and `arrow-compute.pc` files for pkg-config
+respectively, so internally the project also calls:
+
+```meson
+arrow_dep = declare_dependency(...)
+meson.override_dependency('arrow', arrow_dep)
+
+arrow_compute_dep = declare_dependency(...)
+meson.override_dependency('arrow-compute', arrow_compute_dep)
+```
+Note that `meson.override_dependency` was introduced in Meson version
+0.54.0. If your project supports older versions of Meson, you should
+add a condition to only call that function in versions where it is
+available:
+
+```meson
+if meson.version().version_compare('>=0.54.0')
+    meson.override_dependency('arrow', arrow_dep)
+    meson.override_dependency('arrow-compute', arrow_compute_dep)
+endif
+```
+
+In the wrap file, you should then provide the overridden dependency
+in the provides section:
+
+```ini
+[provide]
+dependency_names = arrow, arrow_compute
+```
+
+If the base name of your wrap file matches the name of a dependency,
+then you should *not* include it in the provides section, as Meson
+will do that for your. The aforementioned provide section works for
+the Apache Arrow project because the wrap file is named `arrow-cpp.wrap`,
+but if it were instead just named `arrow.wrap`, then the provides section
+would look like:
+
+```ini
+[provide]
+dependency_names = arrow_compute
+```
```

Comment:
https://github.com/mesonbuild/wrapdb/pull/2345 isn't landing so this paragraph and example are obsolete.  Thanks anyway, though.